github-crawler-core/src/main/kotlin/com/societegenerale/githubcrawler/remote/RemoteBitBucketImpl.kt

```diff
@@ -0,0 +1,341 @@
+package com.societegenerale.githubcrawler.remote
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.societegenerale.githubcrawler.RepositoryConfig
+import com.societegenerale.githubcrawler.model.*
+import com.societegenerale.githubcrawler.model.Author
+import com.societegenerale.githubcrawler.model.Branch
+import com.societegenerale.githubcrawler.model.PullRequest
+import com.societegenerale.githubcrawler.model.Repository
+import com.societegenerale.githubcrawler.model.bitbucket.*
+import com.societegenerale.githubcrawler.model.commit.Commit
+import com.societegenerale.githubcrawler.model.commit.CommitStats
+import com.societegenerale.githubcrawler.model.commit.DetailedCommit
+import com.societegenerale.githubcrawler.model.team.Team
+import com.societegenerale.githubcrawler.model.team.TeamMember
+import feign.*
+import feign.FeignException.errorStatus
+import feign.codec.Decoder
+import feign.codec.ErrorDecoder
+import feign.gson.GsonEncoder
+import feign.httpclient.ApacheHttpClient
+import feign.slf4j.Slf4jLogger
+import okhttp3.Response
+import org.apache.commons.io.IOUtils
+import org.slf4j.LoggerFactory
+import org.springframework.boot.autoconfigure.http.HttpMessageConverters
+import org.springframework.cloud.openfeign.support.ResponseEntityDecoder
+import org.springframework.cloud.openfeign.support.SpringDecoder
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
+import java.io.IOException
+import java.io.StringWriter
+import java.lang.reflect.Type
+import java.util.*
+import java.util.stream.Collectors
+
+@Suppress("TooManyFunctions")
+class RemoteBitBucketImpl @JvmOverloads constructor(
+    val BitBucketUrl: String,
+    val projectName: String = "",
+    val apiKey: String
+) : RemoteSourceControl {
+
+    companion object {
+        const val REPO_LEVEL_CONFIG_FILE = ".BitBucketCrawler"
+    }
+
+    private val internalBitBucketClient: InternalBitBucketClient = Feign.builder()
+        .client(ApacheHttpClient())
+        .encoder(GsonEncoder())
+        .decoder(BitBucketResponseDecoder())
+        .errorDecoder(BitBucketErrorDecoder())
```

Comment:
is the errorDecoder really needed for BitBucket ? it doesn't seem to be, because tests are passing when I remove this line